Feat/render profile info

AritraDey-Dev opened the pull request:
## Fixes Issue #98 

## Changes proposed
- Updated the user form EJS template to point to the new login route.

<!-- List all the proposed changes in your PR -->

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
[Screencast from 2025-02-10 04-11-41.webm](https://github.com/user-attachments/assets/93f3202e-3f60-4492-a122-05acff214751)

sristy17 reviewed (commented):
Great job, @AritraDey-Dev! 🎉 Thank you for submitting your pull request. Your contribution is valuable and we appreciate your efforts to improve our project.<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/sristy17/MindDrive/blob/kwoc-2024/CONTRIBUTING.md)</p>

AritraDey-Dev commented:
@pooranjoyb @sristy17 I have fixed the issue and also i added the whole user data in that session.we can add a database call directly,but i feel that is not a good idea to implement as we are storing that in that session.if you want i can go with that approach.Please share your idea on that.

sristy17 reviewed (changes requested):
Seems good! @pooranjoyb have a look.

Also @AritraDey-Dev  it is not validating the schema of phone number even >10 digits on phone number is working, look into this

AritraDey-Dev commented:
Added `zod` validation schemas for user signup and signin to ensure proper data validation (`controllers/auth.controller.js`).
 
[Screencast from 2025-02-13 14-46-18.webm](https://github.com/user-attachments/assets/7e40ba71-8017-45c1-a1cf-c0e26d265a8b)

pooranjoyb commented on app.js:
Remove unnecessary formatting.

pooranjoyb commented on controllers/auth.controller.js:
Please create a new file called `validator.js` in `utils` and import it here., you shouldn't write validations in the controller file.

pooranjoyb commented on views/pages/profile.ejs:
No need of formatting.

pooranjoyb commented on utils/validator.js:
```diff
@@ -0,0 +1,15 @@
+    username: z.string().min(3, "Username must be at least 3 characters long"),
+    password: z.string().min(6, "Password must be at least 6 characters long"),
+    dob: z.string().refine((date) => !isNaN(Date.parse(date)), { message: "Invalid date format" }),
+    gender: z.enum(["male", "female", "other"]),
```
Validation for `Gender` is not set.

![image](https://github.com/user-attachments/assets/15f06566-24ce-4402-8d4b-19baa914d6da)

pooranjoyb commented on utils/validator.js:
```diff
@@ -0,0 +1,15 @@
+    password: z.string().min(6, "Password must be at least 6 characters long"),
+    dob: z.string().refine((date) => !isNaN(Date.parse(date)), { message: "Invalid date format" }),
+    gender: z.enum(["male", "female", "other"]),
+    fullName: z.string().min(3, "Full Name must be at least 3 characters long"),
```
Not just characters. Validate it for only anything other an Alphabets.

AritraDey-Dev commented:
![Screenshot from 2025-02-17 02-02-23](https://github.com/user-attachments/assets/e55ed964-ac12-4e42-a4cc-18a632e1381d)

pooranjoyb reviewed (approved):
LGTM, Please review and merge @sristy17 if fine.
